infrastructure/src/main/java/org/corfudb/infrastructure/log/InMemoryStreamLog.java

```diff
@@ -193,5 +209,30 @@ public void reset() {
         trimmed.clear();
         // Clearing all data from the cache.
         logCache.clear();
+        logSizeQuota = new ResourceQuota("LogSizeQuota", MAX_MEM_SIZE);
+    }
+
+    @Override
+    public boolean quotaExceeded() {
+        return !logSizeQuota.hasAvailable();
+    }
+
+    @Override
+    public long quotaLimitInBytes() {
+        return logSizeQuota.getLimit();
     }
+
+    /**
+     * startAddress the startAddress of the segment file
+     */
+    @Override
+    public long getSegmentSize(long startAddress) {
+        long size = 0;
+        for(long i = 0; i < RECORDS_PER_LOG_FILE; i++) {
+            LogData data = logCache.get(i + startAddress);
```

Comment:
If startAddress = RECORDS_PER_LOG_FILE/2 then this will be incorrect, since it will also include the "next segment".

I think a cleaner approach would be to pass down both start and end addresses and iterate the map instead of trying to access specific keys. For example, if (ld.getGlobalAddress > start and ld.getGlobalAddress < end).